# Problem Set 1 - Yuqi Gu

## 1 Table 1

### Q1

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import statsmodels.api as sm
import statsmodels.formula.api as smf
import random

In [ ]:
df = pd.read_csv('charitable_giving.csv')
df.head()

,donation_amount,donation_dummy,control,treatment,match_ratio,ratio1,ratio2,ratio3,red_state_dummy,months_since_last_donation,highest_previous_donation,prior_donations_num
0,0.0,0.0,0.0,1.0,1.0,1,0.0,0.0,1.0,19.0,500.0,32.0
1,0.0,0.0,1.0,0.0,0.0,0,0.0,0.0,1.0,29.0,300.0,22.0
2,0.0,0.0,1.0,0.0,0.0,0,0.0,0.0,1.0,3.0,500.0,22.0
3,0.0,0.0,0.0,1.0,3.0,0,0.0,1.0,0.0,4.0,250.0,29.0
4,0.0,0.0,0.0,1.0,2.0,0,1.0,0.0,0.0,8.0,50.0,17.0


In [ ]:
model = smf.ols(formula = 'months_since_last_donation ~ treatment', data = df)
result = model.fit()
print(result.summary())

                                OLS Regression Results                                
Dep. Variable:     months_since_last_donation   R-squared:                       0.000
Model:                                    OLS   Adj. R-squared:                 -0.000
Method:                         Least Squares   F-statistic:                   0.01428
Date:                        Fri, 24 Jan 2025   Prob (F-statistic):              0.905
Time:                                23:06:14   Log-Likelihood:            -1.9585e+05
No. Observations:                       50082   AIC:                         3.917e+05
Df Residuals:                           50080   BIC:                         3.917e+05
Df Model:                                   1                                         
Covariance Type:                    nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------

In [ ]:
13.012-12.998

0.014000000000001123

From the OLS Regression Results above, we observe that the intercept is 12.9981, which means the expected number of months since last donation of control group is 12.9981 months. For the treatment coefficient, we find that it is 0.0137, which means that on average, individuals in the treatment group have been donors for 0.0137 months longer than individuals in the control group.

Furthermore, we find that the regression-based result is very similar to the group-specific means reported in Table 1 of the paper. To be specific, 12.9981 in the result is roughly the same as 12.998 reported in the Table 1. Also, we find 13.012-12.998 = 0.014 based on the Table 1, which is roughly the same as 0.0137 from the regression-based result.

### Q2

The difference in “Number of month since last donation” between treatment and control is not statistically significant (at the usual 95% confidence level). This is because [-0.211, 0.238] contains 0 and the p-value is 0.905, which is larger than 0.05. Thus, there is no significant effect of treatment on the number of month since last donation. This result is we expected. This is because the experiment should be randomized, which means the treatment and control group should be randomly assigned to compare their act of donating. Also, we observe that the difference in “Number of month since last donation” between treatment and control group is very small based on the Table 1.

### Q3

The treatment and control groups show remarkably similar averages across most variables, including the data from census demographics, state and county, etc., indicating that randomization was effective in ensuring comparable groups. The similarity between treatment and control groups supports the validity of the randomization process, ensuring that differences in outcomes can be attributed to the treatment rather than pre-existing differences.

## 2 Response rate regressions

### Q1

In [ ]:
model2 = smf.ols(formula = 'donation_dummy ~ treatment', data = df)
result2 = model2.fit()
print(result2.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     9.618
Date:                Fri, 24 Jan 2025   Prob (F-statistic):            0.00193
Time:                        23:06:20   Log-Likelihood:                 26630.
No. Observations:               50083   AIC:                        -5.326e+04
Df Residuals:                   50081   BIC:                        -5.324e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0179      0.001     16.225      0.0

In [ ]:
0.022-0.018

0.004

Based on the model summary, we find that the intercept is 0.0179, which means that the expected response rate for the control group is 1.79%. The treatment coefficient is 0.0042, which indicates that the response rate for the treatment group is 0.42 percentage points higher than the control group.

Based on the first row of Table 2a, we find that the response rate of control group is 0.018, which is close to the intercept 0.0179 and the difference between the treatment and control group is 0.022-0.018=0.004, which is roughly the same as 0.0042.


### Q2

In [ ]:
model3 = smf.ols(formula = 'donation_dummy ~ ratio1+ratio2+ratio3', data = df)
result3 = model3.fit()
print(result3.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     3.665
Date:                Fri, 24 Jan 2025   Prob (F-statistic):             0.0118
Time:                        23:06:22   Log-Likelihood:                 26630.
No. Observations:               50083   AIC:                        -5.325e+04
Df Residuals:                   50079   BIC:                        -5.322e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0179      0.001     16.225      0.0

The intercept is 0.0179, which indicates that the expected response rate for the control group is 1.79%.

The coefficient for 1:1 match ratio is 0.0029, which means that the response rate for the 1:1 match ratio is 0.29 percentage points higher compared to the control group.

The coefficient for 2:1 match ratio is 0.0048, which means that the response rate for the 2:1 match ratio is 0.48 percentage points higher compared to the control group.

The coefficient for 3:1 match ratio is 0.0049, which means that the response rate for the 3:1 match ratio is 0.49 percentage points higher compared to the control group.


### Q3

In [ ]:
coef_ratio1 = result3.params['ratio1']
coef_ratio2 = result3.params['ratio2']

diff = coef_ratio1 - coef_ratio2
print(diff)

-0.0018842510217150282


The response rate difference between the 1:1 and 2:1 match ratios is -0.0018842510217150282

### Q4


Based on the regression in Q1, the regression analysis confirms that the treatment group has a significantly higher response rate than the control group, with an increase of approximately 0.42 percentage points. This is because the corresponding p-value is 0.002 < 0.05. Moreover, This is also roughly the same as the difference between the treatment and control group, which is 0.004.

Based on the regression in Q2, we find that different match ratios influence response rates, with higher match ratios (2:1 or 3:1) significantly yielding slightly higher response rate comparing with the control group. However, the response rate for these three ratios are close, which separately are 0.021, 0.023, 0.023, based on the Table 2A. Thus, although the match treatments compared with the control group increase the response rate, we find that larger match ratios—3:1 and 2:1—relative to smaller match ratios (1:1) does not have too much additional impact.

Using matched donations is an effective way to improve donation rates, but the cost-effectiveness of higher match ratios should be carefully considered.

## 3 Response rates in red/blue states

### Q1

In [ ]:
df_blue = df[df['red_state_dummy'] == 0]
modelb = smf.ols(formula = 'donation_dummy ~ treatment', data = df_blue)
resultb = modelb.fit()
print(resultb.summary())


                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.000
Method:                 Least Squares   F-statistic:                    0.3567
Date:                Thu, 23 Jan 2025   Prob (F-statistic):              0.550
Time:                        08:22:07   Log-Likelihood:                 15783.
No. Observations:               29806   AIC:                        -3.156e+04
Df Residuals:                   29804   BIC:                        -3.155e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0200      0.001     14.085      0.0

In [ ]:
df_red = df[df['red_state_dummy'] == 1]
modelr = smf.ols(formula = 'donation_dummy ~ treatment', data = df_red)
resultr = modelr.fit()
print(resultr.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     17.24
Date:                Thu, 23 Jan 2025   Prob (F-statistic):           3.31e-05
Time:                        08:22:39   Log-Likelihood:                 10839.
No. Observations:               20242   AIC:                        -2.167e+04
Df Residuals:                   20240   BIC:                        -2.166e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0146      0.002      8.398      0.0


For blue states, we find the intercept is 0.0200, which indicates that the expected response rate for the control group in blue states is 2.00%.
The treatment coefficient is 0.0010, which means that the treatment increases the response rate by 0.10 percentage points compared with the control group in blue states. The p-value is 0.55, which is smaller than 0.05, and so it is not significant.

For red states, the intercept is 0.0146, which means the expected response rate for the control group in red states is 1.46%.
The treatment coefficient is 0.0088, which indicates that the treatment increases the response rate by 0.88 percentage points compared with the control group in red states. The p-value is 0.000 which is much smaller than 0.05 and so it is significant.


The treatment is more effective in red states. The treatment's impact in red states is 0.88 percentage points, compared to only 0.10 percentage points in blue states. And the effect of treatment compared to control on the response rate is significant in red states.

### Q2


The core issue lies in the fact that the distinction between red and blue states primarily reflects differences in many characteristics like the political preferences of their residents. This fundamental difference should not inherently result in variations in treatment effects, which is why these effects lack a causal interpretation. Within each state, the treatment coefficient reflects the average difference in response rates between the treatment and control groups. However, since states are not randomly assigned as red or blue, this coefficient cannot be interpreted as purely causal. Variations in these coefficients may be influenced by underlying state-level characteristics, such as political culture, socioeconomic conditions, or donor demographics, which correlate with both response rates and the effectiveness of the treatment. For example, it is possible that some activities in red states might influence if the people donate or not. In conclusion, the treatment coefficient does not have a causal interpretation in each of the two regressions. Within states, the treatment coefficient suggests a relationship between treatment and response rates but is influenced by unobserved confounding factors and lacks a strictly causal interpretation.

Furthermore, the observed difference in treatment effects between red states and blue states could be due to many other reasons. Specifically, it might be due to differences in donor characteristics, political or cultural factors specific to red vs. blue states and differences in how the treatment resonates with individuals in each type of state, etc. Since states are not randomly assigned, these differences between red states and blue states exist and so the unobserved factors would possibly influence the result. In conclusion, the difference in the treatment effect between states does not have a causal interpretation. It could indicate that the treatment is more effective in one of the states because of pre-existing differences between red and blue states, not because of anything causal about the treatment itself.







## 4 Response rates and donation amount

### Q1

In [ ]:
model4 = smf.ols(formula = 'donation_amount ~ treatment', data = df)
result4 = model4.fit()
print(result4.summary())

                            OLS Regression Results                            
Dep. Variable:        donation_amount   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     3.461
Date:                Thu, 23 Jan 2025   Prob (F-statistic):             0.0628
Time:                        08:28:11   Log-Likelihood:            -1.7946e+05
No. Observations:               50083   AIC:                         3.589e+05
Df Residuals:                   50081   BIC:                         3.589e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.8133      0.067     12.063      0.0

The intercept is 0.8133, which indicates the expected donation amount in the control group is 0.8133.

The treatment coefficient is 0.1536, which is the difference in donation amount for treatment group vs control group and so on average, treatment group donates $0.1536 more than the control group, but this result is not statistically significant since p-value is 0.063, which is larger than 0.05.

From the article, we find "Individuals were randomly assigned to either a control group or a matching grant treatment group, and within the matching grant treatment group individuals were randomly assigned to different matching grant rates, matching grant maximum amounts, and suggested donation amounts". Thus, it is a randomized experiment. Also, the treatment coefficient indicates that treatment group donates $0.1536 more than the control group, and the p-value is 0.063, which is slight bigger than 0.05 but smaller than 0.1. Thus, it is marginally significant. Thus, the treatment coefficient has a causal interpretation.

### Q2

In [ ]:
don = df[df['donation_dummy'] == 1]
model_don = smf.ols(formula = 'donation_amount ~ treatment', data = don)
result_don = model_don.fit()
print(result_don.summary())

                            OLS Regression Results                            
Dep. Variable:        donation_amount   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.001
Method:                 Least Squares   F-statistic:                    0.3374
Date:                Thu, 23 Jan 2025   Prob (F-statistic):              0.561
Time:                        08:30:57   Log-Likelihood:                -5326.8
No. Observations:                1034   AIC:                         1.066e+04
Df Residuals:                    1032   BIC:                         1.067e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     45.5403      2.423     18.792      0.0

The intercept is 45.5403, which indicates the expected donation amount in the control group is 45.5403 among respondents that made a donation.

The treatment coefficient is -1.6684, which is the difference in donation amount for treatment group vs control group among respondents that made a donation and so on average, treatment group donates $1.6684 less than the control group among respondents that made a donation, but this result is not statistically significant since p-value is 0.561, which is larger than 0.05.

The treatment coefficient does not have a causal interpretation. This is because if the people donates or not is not random but it is chosen since the regression is based on respondents that made a donation. Also, the people who are willing to donate might have an expectation on how much they hope to donate,so it will influence the donation amount besides the treatment. Furthermore, the p-value is 0.561, which is larger than 0.1 and 0.05. Thus, it is not statistically significant and so the difference of the donation amount between the control group and treatment group among the respondents that made a donation does not exist since we should not reject null hypothesis. Thus, there is selection bias when we selecting only the people who donate. The treatment coefficient, therefore, captures the treatment's effect conditional on making a donation, not the overall treatment effect on the entire population.